src/main/java/ch/naviqore/app/controller/RoutingController.java

```diff
@@ -42,9 +41,15 @@ public RoutingController(PublicTransitService service) {
         this.service = service;
     }
 
-    private static void handleConnectionRoutingException(ConnectionRoutingException e) {
+    private static ConnectionResponse handleConnectionRoutingException(ConnectionRoutingException e) {
```

Comment:
I vote against wrapping the responses in DTOs again, as it seems like reinventing the wheel.

Spring already provides built-in strategies for exception handling, and we are already using them. See: [Error Handling for REST with Spring](https://www.baeldung.com/exception-handling-for-rest-with-spring)

I think the better approach is to have more speaking exceptions names and handle them in the already existing `ch.naviqore.app.controllerGlobalExceptionHandler` (`@ControllerAdvice`).